src/inventorymanager.cpp

```diff
@@ -412,9 +415,9 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 
 	/* Modify count according to collected data */
 	count = src_item.count;
-	if (src_can_take_count != -1 && count > src_can_take_count)
+	if (src_can_take_count != -1 && count > src_can_take_count && !caused_by_move_somewhere)
 		count = src_can_take_count;
-	if (dst_can_put_count != -1 && count > dst_can_put_count)
+	if (dst_can_put_count != -1 && count > dst_can_put_count && !caused_by_move_somewhere)
 		count = dst_can_put_count;
```

Comment:
I don't think this code is yet correct, but writing it like this might help you during development:
```C++
	if (!caused_by_move_somewhere) {
		// ??? limit the moved stack size to what's allowed to put and take
		if (src_can_take_count != -1 && count > src_can_take_count)
			count = src_can_take_count;
		if (dst_can_put_count != -1 && count > dst_can_put_count)
			count = dst_can_put_count;
	}
```

Reply:
You were correct. (Hopefully) fixed by https://github.com/minetest/minetest/pull/10930/commits/69fa52c1593abca3b3501d1b95e8bf0db87dbe8f.